skimage/transform/_warps.py

```diff
@@ -84,48 +96,40 @@ def resize(image, output_shape, order=1, mode=None, cval=0, clip=True,
         warn("The default mode, 'constant', will be changed to 'reflect' in "
              "skimage 0.15.")
 
-    rows, cols = output_shape[0], output_shape[1]
-    orig_rows, orig_cols = image.shape[0], image.shape[1]
-
-    row_scale = float(orig_rows) / rows
-    col_scale = float(orig_cols) / cols
-
-    # 3-dimensional interpolation
-    if len(output_shape) == 3 and (image.ndim == 2
-                                   or output_shape[2] != image.shape[2]):
-        ndi_mode = _to_ndimage_mode(mode)
-        dim = output_shape[2]
-        if image.ndim == 2:
-            image = image[:, :, np.newaxis]
-        orig_dim = image.shape[2]
-        dim_scale = float(orig_dim) / dim
-
-        map_rows, map_cols, map_dims = np.mgrid[:rows, :cols, :dim]
-        map_rows = row_scale * (map_rows + 0.5) - 0.5
-        map_cols = col_scale * (map_cols + 0.5) - 0.5
-        map_dims = dim_scale * (map_dims + 0.5) - 0.5
-
-        coord_map = np.array([map_rows, map_cols, map_dims])
-
-        image = convert_to_float(image, preserve_range)
-
-        out = ndi.map_coordinates(image, coord_map, order=order,
-                                  mode=ndi_mode, cval=cval)
-
-        _clip_warp_output(image, out, order, mode, cval, clip)
-
-    else:  # 2-dimensional interpolation
-
+    output_shape = np.asarray(output_shape)
+    ndim_out = len(output_shape)
+    input_shape = image.shape
+    if ndim_out > image.ndim:
+        # append dimensions to input_shape
```

Comment:
How did you decide to append instead of prepend? This is the opposite of numpy broadcasting rules, so I would want a justification noted in a comment here...